packages/playwright-test/src/loader.ts

```diff
@@ -131,6 +131,7 @@ export class Loader {
     this._fullConfig.workers = takeFirst(this._configOverrides.workers, config.workers, baseFullConfig.workers);
     this._fullConfig.webServer = takeFirst(this._configOverrides.webServer, config.webServer, baseFullConfig.webServer);
     this._fullConfig._plugins = takeFirst(this._configOverrides.plugins, config.plugins, baseFullConfig._plugins);
+    this._fullConfig.metadata = mergeObjects(this._configOverrides.metadata, mergeObjects(config.metadata, baseFullConfig.metadata));
```

Comment:
this._configOverrides can't override the metadata, so we don't need this line. My recent patches make it clear via explicit type for the overrides.